src/openms/include/OpenMS/ANALYSIS/ID/PeptideIndexing.h

```diff
@@ -677,8 +692,162 @@ namespace OpenMS
+
+     template<typename T>
+     bool findDecoyString_(FASTAContainer<T>& proteins,
+                                std::string& decoy_string,
+                                bool& is_prefix)
+     {
+       // Common decoy strings in fasta files
+       // Note: decoy prefixes/suffices must be provided in lower case and must only contain alphanumeric characters
+        DecoyStringToAffixCount decoy_count = {{"decoy", {0,0}}, {"reverse", {0,0}},
+                                                          {"rev", {0,0}}, {"dev", {0,0}},
+                                                          {"iddecoy", {0,0}}, {"xxx", {0,0}},
+                                                          {"shuffled", {0,0}}, {"shuffle", {0,0}},
+                                                          {"pseudo", {0,0}}, {"random", {0, 0}}};
+       // Map case insensitive strings back to original case (as used in fasta)
+       CaseInsensitiveToCaseSensitiveDecoy decoy_case_sensitive;
+
+       // Setup regex vector, sort descending
+       std::vector<std::string> prefix_regex;
+       for (const auto& pair : decoy_count)
+       {
+         prefix_regex.push_back(pair.first);
+       }
+       std::sort(prefix_regex.rbegin(), prefix_regex.rend());
+
+       // Setup regex string -> create prefix- and suffix pattern to match to
+       std::string regexstr_prefix = "^(";
+       for (const std::string &prefix : prefix_regex)
+       {
+         regexstr_prefix += prefix + "|";
+       }
+       regexstr_prefix = regexstr_prefix.substr(0, regexstr_prefix.length() - 1); // remove last '|'
+       regexstr_prefix += ")";
+       // suffix: remove leading '^', add latter '$'
+       std::string regexstr_suffix(regexstr_prefix.begin() + 1, regexstr_prefix.end());
+       regexstr_suffix.insert(regexstr_suffix.end(), '$');
+
+       boost::regex pattern_prefix(regexstr_prefix);
+       boost::regex pattern_suffix(regexstr_suffix);
+
+       int all_prefix_occur(0), all_suffix_occur(0), all_proteins_count(0);
+       const size_t PROTEIN_CACHE_SIZE = 4e5;
+
+       while (true)
+       {
+         proteins.cacheChunk(PROTEIN_CACHE_SIZE);
+         if (!proteins.activateCache()) break;
+
+         auto prot_count = (SignedSize) proteins.chunkSize();
+         all_proteins_count += prot_count;
+
+         {
+           for (SignedSize i = 0; i < prot_count; ++i)
+           {
+             String seq = proteins.chunkAt(i).identifier;
+             String seq_lower = seq;
+             seq_lower.toLower();
+
+             // remove all non-alphanumeric characters
+             seq_lower.erase(std::remove_if(seq_lower.begin(), seq_lower.end(),
+                                            [](char c) { return !std::isalpha(c); }), seq_lower.end());
+
+
+             boost::smatch sm;
+             // search for prefix
+             bool found_prefix = boost::regex_search(seq_lower, sm, pattern_prefix);
```

Comment:
regex is fine, since it solves the rev/reverse problem, I guess.
But I've seen regex queries take longer than the actual peptideindexing procedure. So could you please provide some numbers on how long this takes on a larger FASTA file? Thanks

Reply:
-- done [took 2.67 s (CPU), 2.68 s (Wall)] -- 
Using suffix decoy string 'rev'
Mapping 4974 peptides to 173450 proteins.

Ran on a thinkpad t440s with an i54200U -> 2,6 GHz max 
4 Gig Ram

Might be faster if I close chromium + Clion.